src/main/scala/edu/ie3/simona/model/participant/evcs/EvcsModel.scala

```diff
@@ -96,16 +96,20 @@ class EvcsModel private (
       receivedData: Seq[Data],
       nodalVoltage: Dimensionless,
   ): EvcsState = {
-    receivedData
-      .collectFirst { case evData: ArrivingEvs =>
-        evData
-      }
-      .map(newData =>
-        state.copy(
-          state.evs ++ newData.arrivals
-        )
-      )
-      .getOrElse(state)
+    val arrivals = receivedData.collect { case evData: ArrivingEvs =>
```

Comment:
To be honest, I wonder why the receivedData Message does not only contain the evs that arrives at this tick. Is there some error somewhere else? Maybe you have some idea @sebastian-peter

Reply:
Let's look into this together next week... I suspect something unexpected is going wrong and we're looking at the wrong spot